File: src/main/java/com/paragon/client/systems/module/impl/client/HUD.java
package com.paragon.client.systems.module.impl.client;

import com.paragon.Paragon;
import com.paragon.api.util.render.ColourUtil;
import com.paragon.api.util.render.TextRenderer;
import com.paragon.client.systems.module.Module;
import com.paragon.client.systems.module.ModuleCategory;
import com.paragon.client.systems.module.hud.HUDModule;
import com.paragon.client.systems.module.settings.impl.BooleanSetting;
import com.paragon.client.systems.module.settings.impl.ModeSetting;
import com.paragon.client.systems.module.settings.impl.NumberSetting;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.ScaledResolution;
import net.minecraft.util.text.TextFormatting;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class HUD extends Module implements TextRenderer {

    public HUD() {
        super("HUD", ModuleCategory.CLIENT, "Render the client's HUD on screen");
    }

    @Override
    public void onRender2D() {
        Paragon.INSTANCE.getModuleManager().getHUDModules().forEach(hudModule -> {
            if (hudModule.isEnabled()) {
                hudModule.render();
            }
        });
    }
}

File: src/main/kotlin/com/paragon/impl/module/hud/impl/HUD.kt
package com.paragon.impl.module.hud.impl

import com.paragon.Paragon
import com.paragon.impl.module.Module
import com.paragon.impl.module.hud.HUDEditorGUI
import com.paragon.impl.module.hud.HUDModule
import com.paragon.impl.module.Category
import com.paragon.util.render.RenderUtil

/**
 * @author Surge
 */
object HUD : Module("HUD", Category.HUD, "Render the client's HUD on screen") {

    override fun onRender2D() {
        if (minecraft.currentScreen !is HUDEditorGUI) {
            Paragon.INSTANCE.moduleManager.getModulesThroughPredicate {
                it is HUDModule && it.animation.getAnimationFactor() > 0
            }.forEach {
                it as HUDModule

                RenderUtil.scaleTo(it.x + (it.width / 2), it.y + (it.height / 2), 0f, it.animation.getAnimationFactor(), it.animation.getAnimationFactor(), 0.0) {
                    it.render()
                }
            }
        }
    }

}